[#6495] Allow for effects to adjust the bloodied threshold

krbz999 opened the pull request:
Stores a numeric value in `CharacterData/NPCData#attributes.bloodiedThreshold` during the base data prep cycle, grabbing the initial value from the CONFIG, which has been adjusted to be 50 instead of 0.5 for simplicity.

Adjusted to compare HP percentage using `attributes.hp.pct` instead of a calculation in the `updateBloodied` method using `hp.effectiveMax` (unsure why this was the way it was).

To allow for the edge case of "bloodied when missing *any* hit points" there's a comparison to `hp.pct === 100` such that an effect can be applied with `system.attributes.bloodiedThreshold | OVERRIDE | 100` to satisfy this requirement.

Closes #6495.

krbz999 committed:
[#6495] Allow for effects to adjust the bloodied threshold.

arbron commented on module/data/actor/templates/attributes.mjs:
```diff
@@ -239,6 +239,16 @@ export default class AttributesFields {
+   * @this {CharacterData|NPCData}
+   */
+  static prepareBaseBloodied() {
+    this.attributes.bloodiedThreshold = CONFIG.DND5E.bloodied.threshold;
```
`bloodiedThreshold: NumberField({ min: 0, max: 100, initial: () => CONFIG.DND5E.bloodied.threshold, persisted: false });` perhaps?

krbz999 replied:
yes, could be simplified a lot with v14.

krbz999 committed:
Rework to use non-persisted field

Fyorl committed:
Style pass

 - Moved bloodiedThreshold under hp, renamed to just bloodied.
 - Some ancillary cleaning:
    - Fixed vehicles being able to become bloodied.
    - Fixed some uses of uncached settings.
